src/rtp/srtp.rs

```diff
@@ -36,71 +111,83 @@ pub struct SrtpContext {
     ) -> Vec<u8> {
-        let iv = self.rtp.salt.rtp_iv(*header.ssrc, srtp_index);
-
         let hlen = header.header_len;
         let input = &buf[hlen..];
+
         assert!(
             input.len() % SRTP_BLOCK_SIZE == 0,
             "RTP body should be padded to 16 byte block size, {header:?} with body length {} was not", input.len()
         );
 
-        let mut output = vec![0_u8; buf.len() + SRTP_HMAC_LEN];
-        self.rtp
-            .enc
-            .encrypt(&iv, input, &mut output[hlen..])
-            .expect("rtp encrypt");
+        match &mut self.rtp {
+            Derived::Aes128CmSha1_80 {
+                hmac, salt, enc, ..
+            } => {
+                // SRTP layout
+                // [header, [rtp, (padding + pad_count)], hmac]
+
+                //     0                   1                   2                   3
+                //     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+                //    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<+
+                //    |V=2|P|X|  CC   |M|     PT      |       sequence number         | |
+                //    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
+                //    |                           timestamp                           | |
+                //    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
+                //    |           synchronization source (SSRC) identifier            | |
+                //    +=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+ |
+                //    |            contributing source (CSRC) identifiers             | |
+                //    |                               ....                            | |
+                //    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
+                //    |                   RTP extension (OPTIONAL)                    | |
+                //  +>+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
+                //  | |                          payload  ...                         | |
+                //  | |                               +-------------------------------+ |
+                //  | |                               | RTP padding   | RTP pad count | |
+                //  +>+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<+
+                //  | ~                     SRTP MKI (OPTIONAL)                       ~ |
+                //  | +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
+                //  | :                 authentication tag (RECOMMENDED)              : |
+                //  | +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ |
+                //  |                                                                   |
+                //  +- Encrypted Portion*                      Authenticated Portion ---+
```

Comment:
Is this layout the same regardless of crypto. Can it stay where it was?

Reply:
It is, but there's a mention around HMAC above. I can shuffle things around